pkgs/stdenv/darwin/make-bootstrap-tools.nix

```diff
@@ -1,21 +1,5 @@
 { pkgspath ? ../../.., test-pkgspath ? pkgspath
 , localSystem ? { system = builtins.currentSystem; }
-# Specify the desired LLVM version in an overlay to avoid the use of
-# mismatching versions.
-#
-# The llvmPackages that we take things (clang, libc++ and such) from
-# is specified explicitly to be llvmPackages_11 to keep the
-# bootstrap-tools stable.  However, tools like otool,
-# install_name_tool and strip are taken straight from stdenv.cc,
-# which, after the bump, is a different LLVM version altogether.
-#
-# The original intent was that bootstrap-tools specified LLVM 11
-# exhaustively but it didn't. That should be rectified with this
-# PR. As to why stick with 11? That's just to keep the
-# bootstrap-tools unchanged.
-#
-# https://github.com/NixOS/nixpkgs/pull/267058/files#r1390889848
-, overlays ? [(self: super: { llvmPackages = super.llvmPackages_11; })]
```

Comment:
This unpins the LLVM used to generate the bootstrap-tools altogether. That means you'll end up with a different bootstrap-tools as soon as Nixpkgs bumps the default LLVM version. This means rebuilding the world on Darwin and the churn this causes too much churn for Darwin Hydra.

I'm fine with bumping the LLVM version used in the bootstrap-tools but I disagree with unpinning it altogether.

Reply:
Wait, isn't rebuilding the world expected when one bumps the default LLVM version on darwin?

Reply:
i just tried:
`nix-build  pkgs/top-level/release.nix -A stdenvBootstrapTools.x86_64-apple-darwin`
after running freshbuildtools and that performed no rebuilds. so i think i am building this correctly.

it also seems that the bootstrap-tools used in the actual bootstrap haven't been updated in a long time. 2020, 2021. i just don't see what the benefit of pinning the compiler. at the very least it makes it harder to test if someone is making compiler changes thinking that just building freshBootstrapTools will test the pinned version, not realizing that the overrides is passing in an empty `[]` and thus not building with the pinned compiler. commit the change and then realize that the tools are all broken on staging-next.

Reply:
Yes, there should be no rebuild in this PR, since there is no rebootstrap (as in #209462). In the same spirit, pinning to `llvmPackages_16` would not help when bumping the default LLVM version to 17/18.

Reply:
> `freshBootstrapTools` is supposed to be an easy way to build the bootstrap-tools as currently used

No, this could never be the case, since bootstrap files also contain `pcre`, `zlib`, `bzip2`, `mpfr`, etc., which could be bumped more often. Even for GCC on Linux,

- The `x86_64-linux` bootstrap files have GCC 9, before the refresh last month.
- The `powerpc64le-linux` bootstrap files on master have GCC 10, while `freshBootstrapTools` gives GCC 13.
- The `aarch64-linux` ones have GCC 12 (and are built by me!).

Instead, I consider `freshBootstrapTools` as an easy way to **refresh** the bootstrap tools.

Reply:
You're right, @wegank. I guess I'm trying to justify how it used to be historically.

This does highlight to me that it's currently highly inconvenient to try to reproduce the bootstrap-tools. Surely we want that to be more transparent?

Reply:
Certainly. It is easy to reproduce the bootstrap files for `x86_64-linux`, by

```bash
$ nix build github:NixOS/nixpkgs/<revision>#freshBootstrapTools.bootstrapFiles.{bootstrapTools,busybox} --system x86_64-linux
```

with the revision specified in the metadata:

https://github.com/NixOS/nixpkgs/blob/ccab9e21200bf766affe161ece719441c35e82a3/pkgs/stdenv/linux/bootstrap-files/x86_64-unknown-linux-gnu.nix#L1-L10

We can note down the same thing for the next Darwin bootstrap files update, too.

Reply:
FWIW: https://github.com/NixOS/nixpkgs/pull/295767  workaround to fix stdenvBootstrapTools on staging-next.